planner/core/rule_build_key_info.go

```diff
@@ -273,8 +278,10 @@ func (is *LogicalIndexScan) BuildKeyInfo(selfSchema *expression.Schema, childSch
 		if path.IsTablePath() {
 			continue
 		}
-		if newKey := checkIndexCanBeKey(path.Index, is.Columns, selfSchema); newKey != nil {
+		if uniqueKey, newKey := checkIndexCanBeKey(path.Index, is.Columns, selfSchema); newKey != nil {
 			selfSchema.Keys = append(selfSchema.Keys, newKey)
+		} else if uniqueKey != nil {
+			selfSchema.UniqueKeys = append(selfSchema.UniqueKeys, uniqueKey)
```

Comment:
Besides from `DataSource` and `LogicalIndexScan`, there are other operators we'd better take care as well, such as `LogicalProjection`, `LogicalAggregation`, etc. 

I noticed that, there is a bug in `LogicalAggregation::BuildKeyInfo()`: it does not check the NotNull flag for the group by columns before appending them into `Keys`. Can you please fix it as well? Or you can fix it in a separate PR, along with handling the remained operators.

Reply:
Could i fix it in a separate PR? ^_^